### PR TITLE
Fix hash code for AliasFilter

### PR DESCRIPTION
**Pull request by jasontedor**
This commit fixes the hash code for AliasFilter as the previous implementation was neglecting to take into consideration the fact that the aliases field is an array and thus a deep hash code of it should be computed rather than a shallow hash code on the reference.

Relates #23994

### CODE REVIEW

File: core/src/test/java/org/elasticsearch/search/internal/AliasFilterTests.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
+package org.elasticsearch.search.internal;
```
Comment (abeyad) [inline core/src/test/java/org/elasticsearch/search/internal/AliasFilterTests.java]: License header above?

**Review by abeyad — APPROVED** [review]
LGTM, left a minor comment about the license header, feel free to push thereafter.

File: core/src/test/java/org/elasticsearch/search/internal/AliasFilterTests.java (hunk 1 above)
Comment (jasontedor) [inline core/src/test/java/org/elasticsearch/search/internal/AliasFilterTests.java]: Thanks, I pushed 041072ac54451e2dbce41936478404d48d9fc7f0.

Comment (abeyad) [inline core/src/test/java/org/elasticsearch/search/internal/AliasFilterTests.java]: 👍

**Comment by jasontedor** [comment]
Thanks @abeyad.

**Pull request merged**